test/unit-test/test_app_btc_common.c

```diff
@@ -986,35 +986,177 @@ print(hashlib.sha256(b''.join(msg)).hexdigest())
 
     uint8_t hash[SHA256_LEN];
 
-    assert_true(btc_common_multisig_hash(BTCCoin_BTC, &multisig, keypath, 4, hash));
+    assert_true(btc_common_multisig_hash(BTCCoin_BTC, &multisig, false, keypath, 4, hash));
     assert_memory_equal(
         hash,
         "\xb0\x26\x7f\xbb\x26\xba\x0e\x74\xba\xd8\x25\xc9\x87\x94\x9f\x58\xba\x22\xaa\x75\xf6\x3b"
         "\x53\x99\x86\xdd\x93\x76\x07\xbb\x4d\xc3",
         sizeof(hash));
 
-    assert_true(btc_common_multisig_hash(BTCCoin_TBTC, &multisig, keypath, 4, hash));
+    assert_true(btc_common_multisig_hash(BTCCoin_TBTC, &multisig, false, keypath, 4, hash));
     assert_memory_equal(
         hash,
         "\x38\x00\xcb\x87\xa1\xe3\x46\xeb\x4a\x61\xe2\x5c\x47\x75\xe6\x63\xf6\x13\x09\x0a\xa2\xbf"
         "\x3f\xdd\xb0\x57\x46\x2d\x17\x4b\x56\xef",
         sizeof(hash));
 
-    assert_true(btc_common_multisig_hash(BTCCoin_LTC, &multisig, keypath, 4, hash));
+    assert_true(btc_common_multisig_hash(BTCCoin_LTC, &multisig, false, keypath, 4, hash));
     assert_memory_equal(
         hash,
         "\x6c\xf1\x81\xd3\xe1\x31\xea\xfe\xfd\x42\x58\x08\x4e\x5e\x48\x36\x6a\x32\xd5\x9b\xe8\x0a"
         "\x0a\xfb\x13\x34\x55\x89\x29\x4c\xcf\x2d",
         sizeof(hash));
 
-    assert_true(btc_common_multisig_hash(BTCCoin_TLTC, &multisig, keypath, 4, hash));
+    assert_true(btc_common_multisig_hash(BTCCoin_TLTC, &multisig, false, keypath, 4, hash));
     assert_memory_equal(
         hash,
         "\x0e\x5e\xe1\xd1\x8a\x74\xd2\x2c\xf7\xe3\x25\x5a\x35\x29\xb9\xa4\x53\xe9\xb0\x80\x00\x5c"
         "\xa0\xbd\x88\x6f\x6d\xec\xf9\xe4\xb8\x45",
         sizeof(hash));
 
-    assert_false(btc_common_multisig_hash(_BTCCoin_MAX + 1, &multisig, keypath, 4, hash));
+    assert_false(btc_common_multisig_hash(_BTCCoin_MAX + 1, &multisig, false, keypath, 4, hash));
+
+    // Test that the hash is correct, and the same for all xpubs permutations if xpubs sort is
+    // enabled.
+
+    // clang-format off
+    const char* permutations[24][4] = {
```

Comment:
Feels like a loop permutating a `const char* xpubs[4]` might've been better. It's just hard to spot where each entry is now and also when updating this in the future, if ever. I think it's easy to make a mistake in the test data which leads to unnecessary debuging: is it my test data or the code I wrote that fails the test?

Reply:
i generated the test data with `itertools.permuate(xpubs, 4)` :P 

didn't feel like spending time writing permutation code in C.

Reply:
hehe. nice. a quick comment for future self and others about itertools.permutate would go a long way imho.

Reply:
Done